astropy/io/fits/convenience.py

```diff
@@ -188,6 +188,7 @@ def getdata(filename, *args, **kwargs):
     data = hdu.data
     if data is None and extidx == 0:
         try:
+            hdulist, extidx = _getext(filename, mode, ext=1, *args, **kwargs)
```

Comment:
it is slightly less efficient though if you only have one extension as you are loading the first one twice, I didn't consider changing that worthwhile but it could be done